woocommerce/woocommerce-gutenberg-products-block
# Rename js/components to js/editor-components

Fixes #2954

This PR renames our editor components (aka `block-components`) so it's clear they are for use in the editor. 

Changes:

- source folder rename  `js/components` => `js/editor-components`
- alias rename `@woocommerce/block-components` => `@woocommerce/editor-components`
- fixed up mentions in docs
- fixed jest config and relative paths in jest tests
- fixed reference in typescript config

### How to test the changes in this Pull Request:

1. Rebuild everything.
2. Smoke test all the blocks - there's a lot of impact here!
3. (Thanks in advance for mega testing.)

Since there's a lot of impact here, feel free to start a checklist of blocks or components to test, so we can tag-team it.

Also test storybook - confirm all relevant stories are working correctly.

### Changelog

> Dev: Renamed `@woocommerce/block-components` alias `@woocommerce/editor-components`.
